AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -2049,6 +2052,67 @@ - (BOOL)supportsRangeManagedInterfaceState
   return ASHierarchyStateIncludesRangeManaged(_hierarchyState);
 }
 
+- (id<ASTraitCollection>)traitCollection
+{
+  ASDN::MutexLocker l(_propertyLock);
+  if (_traitCollection == nil) {
+    _traitCollection = [[ASDisplayTraitCollection alloc] initWithTraits:_internalTraits];
```

Comment:
Yes we definitely need it to be custom. I believe this was a stub with the intention of future extension. The problem to consider is how we allow subclassing but also returning a custom struct that can be used for ASEnvironment down propagation.